### DISCUSSION TITLE
FASTLANE_SESSION expires in 30 mins

### DISCUSSION THREAD

**Discussion by kpowelltech** [fastlane/fastlane#21842] [Q&A]
I am using a FASTLANE_SESSION ENV variable in my CI (Bitrise) to manage hundreds of apps every month. Previously, the session would last for 30 days, but around 1.5 years ago, it was reduced to just 30 minutes, which caused several build failures. To overcome this issue, I created a script that generates a new session cookie every 30 minutes and uses the Bitrise API to refresh the ENV variable in Bitrise. However, I still experience build failures due to an expired session token.

After reading discussions from https://github.com/fastlane/fastlane/pull/14525 and https://github.com/fastlane/fastlane/issues/14305, I was hoping the fix would restore the 30-day session expiration. Unfortunately, it does not work for me. I'm looking for any insights on why this could be happening.

Here's a brief overview of my process:

Fastlane version = 2.219.0
I use fastlane spaceauth to generate a session cookie locally every 30 minutes.
The session cookie contains myacinfo, DES5xxx, and dqsid cookies.
I reformat the cookie and pass it to the Bitrise API to upload. Before passing it to a cURL command, I have to reformat the cookie.
The new cookie is stored in the appropriate ENV variable in Bitrise.
Note that I have to update 11 session cookies for 11 App Store Connect accounts since I manage more than 1,000 apps.
I thought that the DES5xxx in the cookie would extend the session to 30 days, but it seems like I'm missing something.

**Comment by ykhandelwal913** [comment]
@kpowelltech were you able to get any workaround for this? we are also hitting the same problem of 30min expiration.
